pkg/ccl/backupccl/restore_online.go

```diff
@@ -537,7 +547,11 @@ func (r *restoreResumer) maybeWriteDownloadJob(
 	return execConfig.InternalDB.DescsTxn(ctx, func(
 		ctx context.Context, txn descs.Txn,
 	) error {
-		_, err := execConfig.JobRegistry.CreateJobWithTxn(ctx, downloadJobRecord, r.job.ID()+1, txn)
+		downloadJobID := r.job.ID() + 1
+		_, err := execConfig.JobRegistry.CreateJobWithTxn(ctx, downloadJobRecord, downloadJobID, txn)
+		if err == nil {
```

Comment:
do you need this branch?

Reply:
My thought process was that in the event that the `CreateJobWithTxn` failed, then there would be no download job started, so the resumer shouldn't store that job ID.

Reply:
a more idiomatic approach would be:
```
if _, err := CreateJobWithTxn(); err != nil {
  return err
}
r.DownloadJobID = downloadJobID
return nil
```